ibazel/main.go

```diff
@@ -94,27 +94,30 @@ func main() {
 		return
 	}
 
+	// Create a timer objet to use for debouncing. When the timer is created, do
+	// not start the timer but instead start the timer when an event that builds
+	// should be run.  This has the added benefit of kicking things off by
+	// sending an event to make the first run happen.
+	debouncer := time.NewTimer(debounceDuration)
```

Comment:
this debounce logic seems sound.  Here's another approach which might have slightly less complex code with slightly more variability of debounce time.  Your way works totally fine, though.

```go
tick := time.Tick(debounceTime / 2)
state := "RUN"
for {
  select {
    case <-event:
      state = "EVENT"
    case <-tick:
      switch state {
        case "EVENT":
          state = "RUN"
        case "RUN":
          runCommand()
          state = "DONE"
        case "DONE":
          // nothing
      }
  }
}